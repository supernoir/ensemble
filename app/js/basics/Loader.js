import React from 'react';
import intl from 'react-intl-universal';
import { Dimmer, Loader as SemanticLoader } from 'semantic-ui-react';

/**
 * Class Loader
 * component to be shown when a parent is loading
 */
export default class Loader extends React.Component {
	render(){
		return (
			<Dimmer active>
				<SemanticLoader inverted size='big'>{intl.get('app.loading')}</SemanticLoader>
			</Dimmer>
		);
	}
}