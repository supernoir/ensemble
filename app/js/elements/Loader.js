import React from 'react';
import intl from 'react-intl-universal';
import { Dimmer, Loader as SemanticLoader } from 'semantic-ui-react';

export default class Loader extends React.Component {
	render(){
		return (
			<Dimmer active>
				<SemanticLoader inverted size='big'>{intl.get('app.loading')}</SemanticLoader>
			</Dimmer>
		);
	}
}