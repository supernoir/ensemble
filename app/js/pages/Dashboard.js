import React from 'react';
import intl from 'react-intl-universal';
import moment from 'moment';
import { Container, Segment, Header, Divider, Card, Feed } from 'semantic-ui-react';

export default class Dashboard extends React.Component {
	constructor() {
		super();
		this.state = {
			eventsData: []
		};
	}

	parseEvent = event => {
		switch (event.action) {
			case 'add_project':
				return `You added ${event.ref} to your projects.`;
			case 'add_character':
				return `You added ${event.ref} to your Characters.`;
			case 'edit_project':
			case 'edit_character':
				return `You edited ${event.ref}`;
		}
	};


	componentDidMount() {
		this.props.getEvents();
	}


	render() {
		return (
			<Container>
				<Segment>
					<Header>
						{intl.get('component.dashboard')}
					</Header>
					<Divider />
					<Card.Group>
						<Card
							href="/projects"
							image={'../../public/img/photo-1443188631128-a1b6b1c5c207.jpeg'}
							header={intl.get('entity.projects')}
							meta={`12 Projects available`}
						/>
						<Card
							href="/characters"
							image={'../../public/img/photo-1427805371062-cacdd21273f1.jpeg'}
							header={intl.get('entity.characters')}
							meta={`4 Characters available`}
						/>
					</Card.Group>
					<Header>
						{'I18N: Recent Activity'}
						<Divider />

						{this.props.eventsData
							? this.props.eventsData.map(event => {
								return (
									<Feed>
										<Feed.Event>
											<Feed.Label image={'../../public/img/protouser.png'} />
											<Feed.Content>
												<Feed.Date>{moment(event.timestamp, 'X').format('lll')}</Feed.Date>
												<Feed.Summary>
													{this.parseEvent(event)}
												</Feed.Summary>
											</Feed.Content>
										</Feed.Event>
									</Feed>
								);
							})
							: null}

					</Header>
				</Segment>
			</Container>
		);
	}
}
