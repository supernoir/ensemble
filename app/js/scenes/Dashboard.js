import React from 'react';
import { Link } from 'react-router-dom';
import intl from 'react-intl-universal';
import moment from 'moment';
import { Container, Segment, Header, Divider, Card, Feed } from 'semantic-ui-react';
import Loader from '../layout/Loader';

export default class Dashboard extends React.Component {
	constructor() {
		super();
		this.state = {
			eventsData   : [],
			dashboardData: []
		};
	}

	parseEvent = event => {
		switch (event.action) {
			case 'add_project':
				return intl.get('event.action-addproject', { ref: event.ref });
			case 'add_character':
				return intl.get('event.action-addcharacter', { ref: event.ref });
			case 'edit_project':
				return intl.get('event.action-editproject', { ref: event.ref });
			case 'edit_character':
				return intl.get('event.action-editcharacter', { ref: event.ref });
			case 'delete_project':
				return intl.get('event.action-deletecharacter', { ref: event.ref });
			case 'delete_character':
				return intl.get('event.action-deletecharacter', { ref: event.ref });
		}
	};

	componentDidMount() {
		this.props.getEvents();
		this.props.getDashboardData();
	}

	render() {
		return this.props.loading
			? <Loader loading={this.props.loading} />
			: <Container>
				<Segment>
					<Header>
						{intl.get('component.dashboard')}
					</Header>
					<Divider />
					<Card.Group>
						<Card
							onClick={() => this.props.history.push('/projects')}
							image={'../../public/img/photo-1443188631128-a1b6b1c5c207.jpeg'}
							header={intl.get('entity.projects')}
							meta={intl.get('project.count-available', {
								count: this.props.dashboardData ? this.props.dashboardData.projects : 0
							 })}
						/>
						<Card
							onClick={() => this.props.history.push('/characters')}
							image={'../../public/img/photo-1427805371062-cacdd21273f1.jpeg'}
							header={intl.get('entity.characters')}
							meta={intl.get('character.count-available', {
								count: this.props.dashboardData ? this.props.dashboardData.characters : 0
							})}
						/>
					</Card.Group>
					<Header>
						<Link to="/events">{intl.get('event.recent-activity')}</Link>
						<Divider />

						{this.props.eventsData
							? this.props.eventsData.map(event => {
								return (
									<Feed>
										<Feed.Event>
											<Feed.Label image={'../../public/img/protouser.png'} />
											<Feed.Content>
												<Feed.Date>{moment(event.timestamp, 'X').format('lll')}</Feed.Date>
												<Feed.Summary>
													{this.parseEvent(event)}
												</Feed.Summary>
											</Feed.Content>
										</Feed.Event>
									</Feed>
								);
							})
							: null}

					</Header>
				</Segment>
			</Container>;
	}
}
