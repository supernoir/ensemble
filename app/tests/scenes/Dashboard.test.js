import React from 'react';
import { shallow } from 'enzyme';

// Import Components
import Dashboard from './../../js/scenes/Dashboard';
import Loader from '../../js/basics/Loader';

// Import Mock Data
//import * as mockData from './mocks/mockApiResponse.json';

test('Dashboard -- Snapshot Test', () => {
	const wrapper = shallow(<Dashboard getEvents={() => true} getDashboardData={() => true} />);
	expect(wrapper).toMatchSnapshot();
});
/*
describe('Dashboard', function() {
	it('should render a loader', function() {
		expect(shallow(<Dashboard getEvents={() => true} getDashboardData={() => true} loading={true} />).contains(<Loader />)).toBe(true);
	});
}); */
