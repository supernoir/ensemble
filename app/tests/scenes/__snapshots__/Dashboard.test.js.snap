// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard -- Snapshot Test 1`] = `
<Container>
  <Segment>
    <Header />
    <Divider />
    <CardGroup>
      <Card
        header=""
        href="/projects"
        image="../../public/img/photo-1443188631128-a1b6b1c5c207.jpeg"
        meta="12 Projects available"
      />
      <Card
        header=""
        href="/characters"
        image="../../public/img/photo-1427805371062-cacdd21273f1.jpeg"
        meta="4 Characters available"
      />
    </CardGroup>
    <Header>
      <Link
        replace={false}
        to="/events"
      >
        I18N: Recent Activity
      </Link>
      <Divider />
    </Header>
  </Segment>
</Container>
`;
